### PR TITLE
Feature: Azure Infrastructure Update for Vantiq 1.37

### PR DESCRIPTION
**Pull request by tsyunan-vantiq**
VANTIQ 1.37以降でLLM連携するのに対応する

変更箇所：

- AI Assistant Node Poll追加
- EKS, Terraformバージョンアップ（バージョンアップに伴いdeprecatedパラメータ変更）
- コードCleanUp（`backend.tf`, `provider.tf`に記述する）
- [terraform-docs](https://github.com/terraform-docs/terraform-docs)でREADMEドキュメント追加
- READMEに1.37対応するdeploy.yaml/secrets.yamlプロパティ追加

### CODE REVIEW

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf [hunk 1]
```diff
@@ -10,15 +10,11 @@ resource "azurerm_kubernetes_cluster" "aks-vantiq" {
...
-    enabled = true
-  }
-  kubernetes_version  = var.kubernetes_version
-  node_resource_group = "${var.resource_group_name}-node"
+  kubernetes_version      = var.kubernetes_version
+  node_resource_group     = "${var.resource_group_name}-node"
 
   network_profile {
     dns_service_ip     = var.dns_service_ip
-    docker_bridge_cidr = var.docker_bridge_cidr
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:21]: Will be deprecated
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#docker_bridge_cidr

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf [hunk 2]
```diff
@@ -10,15 +10,11 @@ resource "azurerm_kubernetes_cluster" "aks-vantiq" {
   resource_group_name     = azurerm_resource_group.rg-aks.name
   dns_prefix              = var.aks_cluster_name
   private_cluster_enabled = var.private_cluster_enabled
-  role_based_access_control {
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:13]: Deprecated. Default `true`.
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#role_based_access_control_enabled

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf [hunk 3]
```diff
@@ -45,13 +41,15 @@ resource "azurerm_kubernetes_cluster" "aks-vantiq" {
   }
 
   default_node_pool {
-    name               = "keycloaknp"
-    node_count         = var.keycloak_node_pool_node_count
-    vm_size            = var.keycloak_node_pool_vm_size
-    os_disk_type       = var.keycloak_node_pool_node_ephemeral_os_disk ? "Ephemeral" : null
-    os_disk_size_gb    = var.keycloak_node_pool_node_ephemeral_os_disk ? 64 : null
-    vnet_subnet_id     = var.aks_node_subnet_id
-    availability_zones = var.availability_zones
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:54]: Deprecated.
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#zones

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf [hunk 4]
```diff
@@ -45,13 +41,15 @@ resource "azurerm_kubernetes_cluster" "aks-vantiq" {
...
+    name            = "keycloaknp"
+    node_count      = var.keycloak_node_pool_node_count
+    vm_size         = var.keycloak_node_pool_vm_size
+    os_disk_type    = var.keycloak_node_pool_node_ephemeral_os_disk ? "Ephemeral" : null
+    os_disk_size_gb = var.keycloak_node_pool_node_ephemeral_os_disk ? 64 : null
+    vnet_subnet_id  = var.aks_node_subnet_id
+    zones           = var.availability_zones
+    # Specifies the name of the temporary node pool used to cycle the default node pool for VM resizing
+    temporary_name_for_rotation = "keycktempnp"
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:52]: - https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#temporary_name_for_rotation

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf [hunk 5]
```diff
@@ -74,16 +72,10 @@ resource "azurerm_kubernetes_cluster" "aks-vantiq" {
     }
   }
 
-  addon_profile {
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:77]: Deprecated.

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf [hunk 6]
```diff
@@ -74,16 +72,10 @@ resource "azurerm_kubernetes_cluster" "aks-vantiq" {
...
-    dynamic "oms_agent" {
-      for_each = var.loganalytics_enabled != null ? tolist(["1"]) : []
-      content {
-        enabled                    = var.loganalytics_enabled
-        log_analytics_workspace_id = var.loganalytics_enabled == true ? azurerm_log_analytics_workspace.k8s[0].id : null
-      }
-    }
-    kube_dashboard {
-      enabled = false
+  dynamic "oms_agent" {
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:75]: - https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#oms_agent

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf [hunk 7]
```diff
@@ -99,7 +91,7 @@ resource "azurerm_kubernetes_cluster_node_pool" "vantiqnp" {
   os_disk_size_gb       = var.vantiq_node_pool_node_ephemeral_os_disk ? 64 : null
   node_count            = var.vantiq_node_pool_node_count
   vnet_subnet_id        = var.aks_node_subnet_id
-  availability_zones    = var.availability_zones
+  zones                 = var.availability_zones
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:94]: Deprecated.
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster_node_pool#zones

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf [hunk 8]
```diff
@@ -172,11 +164,29 @@ resource "azurerm_kubernetes_cluster_node_pool" "metricsnp" {
...
 
   node_labels = {
     "vantiq.com/workload-preference" = "compute"
   }
   orchestrator_version = var.kubernetes_version
-}
\ No newline at end of file
+}
+
+resource "azurerm_kubernetes_cluster_node_pool" "vantiqaiasnp" {
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:176]: New node pool.

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/service_principal.tf [hunk 9]
```diff
@@ -15,8 +15,8 @@ resource "azuread_application" "aks" {
 
 # Create Service Principal
 resource "azuread_service_principal" "aks" {
-  count          = var.create_service_principal == false ? 0 : 1
-  application_id = azuread_application.aks[0].application_id
+  count     = var.create_service_principal == false ? 0 : 1
+  client_id = azuread_application.aks[0].client_id
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/service_principal.tf:19]: Deprecated.
- https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/resources/service_principal#client_id

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/variables.tf [hunk 10]
```diff
@@ -41,15 +41,11 @@ variable "dns_service_ip" {
...
-  description = "docker bridge cidr"
-  type        = string
-  default     = "172.17.0.1/16"
-}
+
 variable "load_balancer_sku" {
   description = "load balacer sku. must be standard or higher"
   type        = string
-  default     = "Standard"
+  default     = "standard"
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/variables.tf:48]: Default value should be `standard`.
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#load_balancer_sku

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/opnode/init-script.sh [hunk 11]
```diff
@@ -55,7 +55,18 @@ wget https://github.com/bitnami-labs/sealed-secrets/releases/download/v0.9.5/kub
 install -m 755 kubeseal /usr/local/bin/kubeseal
 
 # install az-cli
-curl -sL https://aka.ms/InstallAzureCLIDeb | bash
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/opnode/init-script.sh:58]: This installation might not work.

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/opnode/init-script.sh [hunk 12]
```diff
@@ -55,7 +55,18 @@ wget https://github.com/bitnami-labs/sealed-secrets/releases/download/v0.9.5/kub
...
+apt-get update
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/opnode/init-script.sh:58]: - Ref: https://learn.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=apt#option-2-step-by-step-installation-instructions

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/storage/storage.tf [hunk 13]
```diff
@@ -15,8 +15,8 @@ resource "azurerm_storage_account" "vantiq" {
   access_tier              = "Hot"
 
   # security
-  enable_https_traffic_only = true
-  allow_blob_public_access  = false
+  enable_https_traffic_only       = true
+  allow_nested_items_to_be_public = false
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/storage/storage.tf:19]: Deprecated.
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#allow_nested_items_to_be_public

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/vpc/snet.tf [hunk 14]
```diff
@@ -4,7 +4,7 @@ resource "azurerm_subnet" "snet-aks-node" {
   resource_group_name                            = azurerm_resource_group.rg-vpc.name
   virtual_network_name                           = azurerm_virtual_network.vnet-vantiq.name
   address_prefixes                               = var.snet_aks_node_address_cidr
-  enforce_private_link_endpoint_network_policies = true
+  private_endpoint_network_policies_enabled = true
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/vpc/snet.tf:7]: Deprecated.
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/subnet#private_endpoint_network_policies_enabled

File: vantiq-cloud-infra-operations/terraform_azure/new/env-template/03_main/main.tf [hunk 15]
```diff
@@ -150,25 +83,28 @@ module "aks" {
...
+  keycloak_node_pool_vm_size                           = module.constants.aks_config.keycloak_node_pool_vm_size
+  keycloak_node_pool_node_count                        = module.constants.aks_config.keycloak_node_pool_node_count
+  keycloak_node_pool_node_ephemeral_os_disk            = module.constants.aks_config.keycloak_node_pool_node_ephemeral_os_disk
+  metrics_node_pool_vm_size                            = module.constants.aks_config.metrics_node_pool_vm_size
+  metrics_node_pool_node_count                         = module.constants.aks_config.metrics_node_pool_node_count
+  metrics_node_pool_node_ephemeral_os_disk             = module.constants.aks_config.metrics_node_pool_node_ephemeral_os_disk
+  vantiq_ai_assistant_node_pool_vm_size                = module.constants.aks_config.vantiq_ai_assistant_node_pool_vm_size
+  vantiq_ai_assistant_node_pool_node_count             = module.constants.aks_config.vantiq_ai_assistant_node_pool_node_count
+  vantiq_ai_assistant_node_pool_node_ephemeral_os_disk = module.constants.aks_config.vantiq_ai_assistant_node_pool_node_ephemeral_os_disk
```
Comment (junanzheng) [inline vantiq-cloud-infra-operations/terraform_azure/new/env-template/03_main/main.tf:107]: `Terraform fmt`

File: vantiq-cloud-infra-operations/terraform_azure/new/env-dev/03_main/main.tf [hunk 16]
```diff
@@ -129,7 +63,6 @@ module "aks" {
   #  vnet_subnet_id = data.terraform_remote_state.network.outputs.vantiq_aks_node_subnet_kubenet_id
 
   # network profile (optional)
-  docker_bridge_cidr = module.constants.aks_config.docker_bridge_cidr
   #  load_balancer_sku =  module.constants.aks_config.load_balancer_sku
```
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/env-dev/03_main/main.tf:66]: Will be deprecated.
-  https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#docker_bridge_cidr

File: vantiq-cloud-infra-operations/terraform_azure/new/env-dev/03_main/main.tf [hunk 17]
```diff
@@ -150,25 +83,28 @@ module "aks" {
...
+  keycloak_node_pool_vm_size                           = module.constants.aks_config.keycloak_node_pool_vm_size
+  keycloak_node_pool_node_count                        = module.constants.aks_config.keycloak_node_pool_node_count
+  keycloak_node_pool_node_ephemeral_os_disk            = module.constants.aks_config.keycloak_node_pool_node_ephemeral_os_disk
+  metrics_node_pool_vm_size                            = module.constants.aks_config.metrics_node_pool_vm_size
+  metrics_node_pool_node_count                         = module.constants.aks_config.metrics_node_pool_node_count
+  metrics_node_pool_node_ephemeral_os_disk             = module.constants.aks_config.metrics_node_pool_node_ephemeral_os_disk
+  vantiq_ai_assistant_node_pool_vm_size                = module.constants.aks_config.vantiq_ai_assistant_node_pool_vm_size
+  vantiq_ai_assistant_node_pool_node_count             = module.constants.aks_config.vantiq_ai_assistant_node_pool_node_count
+  vantiq_ai_assistant_node_pool_node_ephemeral_os_disk = module.constants.aks_config.vantiq_ai_assistant_node_pool_node_ephemeral_os_disk
 }
```
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/env-dev/03_main/main.tf:108]: Using `terraform fmt` for code format.
Automatically format Terraform configuration files (.tf files) according to a defined style guide.

File: vantiq-cloud-infra-operations/terraform_azure/new/env-dev/constants.tf [hunk 18]
```diff
@@ -117,7 +98,6 @@ locals {
     dns_service_ip = "10.1.1.10"
 
     # network profile (optional)
-    docker_bridge_cidr = "172.17.0.1/16"
```
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/env-dev/constants.tf:120]: Will be deprecated.
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#docker_bridge_cidr

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf (hunk 2 above)
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:13]: Deprecated. Default `true`.
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#role_based_access_control_enabled

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf (hunk 1 above)
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:21]: Will be deprecated. `Optional`
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#docker_bridge_cidr

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf [hunk 19]
```diff
@@ -45,13 +41,15 @@ resource "azurerm_kubernetes_cluster" "aks-vantiq" {
...
-    os_disk_size_gb    = var.keycloak_node_pool_node_ephemeral_os_disk ? 64 : null
-    vnet_subnet_id     = var.aks_node_subnet_id
-    availability_zones = var.availability_zones
+    name            = "keycloaknp"
+    node_count      = var.keycloak_node_pool_node_count
+    vm_size         = var.keycloak_node_pool_vm_size
+    os_disk_type    = var.keycloak_node_pool_node_ephemeral_os_disk ? "Ephemeral" : null
+    os_disk_size_gb = var.keycloak_node_pool_node_ephemeral_os_disk ? 64 : null
+    vnet_subnet_id  = var.aks_node_subnet_id
+    zones           = var.availability_zones
```
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:50]: Deprecated.
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#zones

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf (hunk 4 above)
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:52]: - https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#temporary_name_for_rotation

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf (hunk 5 above)
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/aks.tf:77]: Deprecated.

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/service_principal.tf [hunk 20]
```diff
@@ -15,8 +15,8 @@ resource "azuread_application" "aks" {
 
 # Create Service Principal
 resource "azuread_service_principal" "aks" {
-  count          = var.create_service_principal == false ? 0 : 1
-  application_id = azuread_application.aks[0].application_id
```
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/service_principal.tf:19]: Deprecated.
- https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/resources/service_principal#client_id

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/variables.tf (hunk 10 above)
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/aks/variables.tf:48]: Default value should be `standard`.
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#load_balancer_sku

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/opnode/init-script.sh (hunk 11 above)
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/opnode/init-script.sh:58]: This installation might not work.
- https://learn.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=apt#option-2-step-by-step-installation-instructions

File: vantiq-cloud-infra-operations/terraform_azure/new/modules/storage/storage.tf [hunk 21]
```diff
@@ -15,8 +15,8 @@ resource "azurerm_storage_account" "vantiq" {
   access_tier              = "Hot"
 
   # security
-  enable_https_traffic_only = true
-  allow_blob_public_access  = false
```
Comment (tsyunan-vantiq) [inline vantiq-cloud-infra-operations/terraform_azure/new/modules/storage/storage.tf:19]: Deprecated.

**Comment by tsyunan-vantiq** [comment]
@Takahashi-3Yx 
- aiassistantのノードグループ名変更しました。
- 「VANTIQ 1.37バージョン以降インストールする変更箇所」消しました
- 各モジュールにREADMEを追加する：可読性、Moduleメンテナンスのため
- `Terraform fmt`コードフーマットコマンドです: https://github.com/fujitake/vantiq-related/pull/242#discussion_r1562134922
- docker_bridge_cidrの箇所を削除: https://github.com/fujitake/vantiq-related/pull/242#discussion_r1562137440

**Review by fujitake — APPROVED** [review]

**Pull request merged**